### ISSUE TITLE
Train a domain-specific BERT model,get very large loss. 

### ISSUE THREAD

**Issue by wxp16** [NVIDIA/DeepLearningExamples#172]
I used `run_pretraining.py`  to train a domain-specific BERT model from scratch.  Below are some of my parameters: 
- number of gpus = 8
- initial learning rate=1e-4, cos the learning rate is scaled up, for each worker/gpu, the initial learning rate should be 8 * 1e-4
- train batch size=14
- train steps = 1000000
- warmup steps=10000
- use fp16 
- use horovod
- use xla

The loss gets very large at some point, and never drops . Has anyone seen this issue? Thanks
![image](https://user-images.githubusercontent.com/18211419/63602761-3109c180-c596-11e9-8082-cd1328be161e.png)

**Comment by nvcforster** [comment]
Hi, would it be possible to get your exact command line? It is possible this behavior is related to learning rate. Have you tried using smaller learning rates?

**Comment by wxp16** [comment]
> Hi, would it be possible to get your exact command line? It is possible this behavior is related to learning rate. Have you tried using smaller learning rates?

Thanks for your quick reply. Yes, I think this is  due to the learning rate.  I tried another run, but without scaling up the learning rate (i.e. for all 8 workers, the initial learning rate is 1e-4, rather than 8*1e-4), the loss change is quite normal. 

I just wonder to know when do we need to scale up the learning rate. is this related to the train batch size? Any suggestions about this. Thanks

**Comment by nvcforster** [comment]
Thanks for letting us know. It is scaled for you, so you don't have to worry about world size.
https://gitlab-master.nvidia.com/dl/JoC/bert_tf/blob/nvidia-master/run_pretraining.py#L491

**Comment by nvcforster** [comment]
Thanks for letting us know. It is scaled for you, so you don't have to worry about world size.

https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow/LanguageModeling/BERT/run_pretraining.py#L491

**Comment by wxp16** [comment]
> 
> 
> Thanks for letting us know. It is scaled for you, so you don't have to worry about world size.
> 
> https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow/LanguageModeling/BERT/run_pretraining.py#L491
Scaling up will cause error. Looks like we don't need scale up if the batch size is small.

**Comment by nvcforster** [comment]
The idea is that linearly scaling the learning rate based on number of workers (gpus) works for a wide range of workers and learning rate, and the learning rate specified on the command line will be the same for single-gpu and multi-gpu training to make it easier to run the same job at multiple scales. Do you have a specific case where this relationship is failing?
